src/main/java/it/gov/pagopa/payhub/auth/service/user/IamUserInfoDTO2UserInfoMapper.java

```diff
@@ -27,26 +29,48 @@ public IamUserInfoDTO2UserInfoMapper(UsersRepository usersRepository, OperatorsR
 
     @Override
     public UserInfo apply(IamUserInfoDTO iamUserInfoDTO) {
+        if (iamUserInfoDTO.isSystemUser()) {
+            return systemUserMapper(iamUserInfoDTO);
+        }
+        return userInfoMapper(iamUserInfoDTO);
+    }
+
+    private UserInfo systemUserMapper(IamUserInfoDTO iamUserInfoDTO) {
+        return UserInfo.builder()
+          .userId(iamUserInfoDTO.getUserId())
+          .mappedExternalUserId(iamUserInfoDTO.getOrganizationAccess().getOrganizationIpaCode() + WS_USER_SUFFIX)
```

Comment:
instead of read it again each time, could you store this info into a variable?
`iamUserInfoDTO.getOrganizationAccess().getOrganizationIpaCode()`